#485 add basic backoffice application

filippomc opened the pull request:
Closes #485

filippomc commented on applications/osb-portal copy/src/components/workspace/Workspaces.tsx:
```diff
@@ -0,0 +1,238 @@
+import * as React from "react";
```
no, thanks removed

filippomc commented on applications/backoffice/deploy/values.yaml:
```diff
@@ -0,0 +1,20 @@
+harness:
+  subdomain: admin
```
yes

filippomc commented on applications/backoffice/src/config.ts:
```diff
@@ -0,0 +1,4 @@
+
+export const CONFIGURATION = {
+    "appName": "osb-portal"
```
config.ts may be unused

sanjayankur31 commented:
Fixed typos.

How should one test this in a local deployment? `admin.osb.local` doesn't work yet. The build doesn't list this as one of the domains:

```
Starting deploy...                                                                                                                                                           
Helm release osblocal not installed. Installing...                                                                                                                           
NAME: osblocal                                                                                                                                                               
LAST DEPLOYED: Thu May 12 11:05:31 2022                                                                                                                                      
NAMESPACE: osblocal                                                                                                                                                          
STATUS: deployed                                                                                                                                                             
REVISION: 1                                                                                                                                                                  
TEST SUITE: None                                                                                                                                                             
NOTES:                                                                                                                                                                       
1. Domains:                                                                                                                                                                  

  https://osb.local
  https://accounts.osb.local
  https://api.accounts.osb.local
  https://argo.osb.local
  https://common.osb.local
  https://events.osb.local
  https://hub.osb.local
  https://notebooks.osb.local
  https://notifications.osb.local
  https://www.osb.local
  https://volumemanager.osb.local
  https://workflows.osb.local
  https://workspaces.osb.local

```

I had deleted the deployment and done it all again---something else I need to do?

running `harness-deployment` also does not list `admin.osb.local` in the usual places, like here:

```
To test locally, update your hosts file
192.168.49.2    osb.local accounts.osb.local argo.osb.local common.osb.local events.osb.local hub.osb.local notifications.osb.local volumemanager.osb.local workflows.osb.local notebooks.osb.local www.osb.local workspaces.osb.local api.accounts.osb.local
```

I went ahead and updated my hosts file anyway, but then got a 404 at `admin.osb.local` from nginx.

filippomc commented:
> 
> I had deleted the deployment and done it all again---something else I need to do?
> 

Applications are included using the `-i` parameter and dependencies specified on values.yaml file are also included.

I added the dependency also on `harness-deployment ... -i osb-portal -e local` setup, but it will be more efficient if you run `harness-deployment ... -i backoffice -e local` as you'll get a deployment with the backoffice and backends only

sanjayankur31 commented:
Thanks, sounds good. 

Got an npm error while building it now:

```
#11 [base 5/7] RUN npm ci
#11 sha256:3769961d9f262f4511fb9edf162efcd37d5518a57584d2b862e0fff6194fda68
#11 0.673 npm ERR! cipm can only install packages when your package.json and package-lock.json or npm-shrinkwrap.json are in sync. Please update your lock file with `npm install` before continuing.
#11 0.673 npm ERR!
#11 0.673 npm ERR!
#11 0.673 npm ERR! Missing: @mui/icons@^5.7.0
#11 0.673 npm ERR!
#11 0.683
#11 0.683 npm ERR! A complete log of this run can be found in:
#11 0.683 npm ERR!     /root/.npm/_logs/2022-05-12T13_21_22_434Z-debug.log
#11 ERROR: executor failed running [/bin/sh -c npm ci]: exit code: 1
------
 > [base 5/7] RUN npm ci:
------
executor failed running [/bin/sh -c npm ci]: exit code: 1
Build [osb/backoffice] failed: exit status 1. Docker build ran into internal error. Please retry.

```

Also get it when I run `npm install`:

```
npm install
npm WARN old lockfile
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@mui%2ficons - Not found
npm ERR! 404
npm ERR! 404  '@mui/icons@^5.7.0' is not in this registry.
npm ERR! 404
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/asinha/.npm/_logs/2022-05-12T13_22_27_815Z-debug-0.log
```

Are we looking for `@material-ui/icons`?

https://v4.mui.com/components/icons/#icons or some other module?

filippomc commented:
Fixed. Using material-ui 5 here, now called mui

sanjayankur31 commented:
Thanks. I made a few tweaks to make tslint happy, otherwise the local deployment was failing to build.

Should be good to go now. We can make further fixes as I work on adding the necessary bits.

sanjayankur31 commented:
Checked with deployment, I get the homepage now at admin.osb.local. Let's merge this and continue adding bits to develop via individual PRs.
